### PR TITLE
More logs on node freeze

### PR DESCRIPTION
**Pull request by folex**

### CODE REVIEW

File: particle-node/src/futures_handle.rs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+use async_std::task::JoinHandle;
+use futures::future::{BoxFuture, FusedFuture};
+use futures::stream::Fuse;
+use futures::{FutureExt, Stream};
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+pub struct FuturesHandle {
```
Comment (michaelvoronov) [inline particle-node/src/futures_handle.rs:25]: Maybe rename it to smth like `NetworkHandles`/`NetworkFutures`/`NodeFutures`? `FuturesHandles` sounds too general for me.

File: deploy/fluence.yml [hunk 2]
```diff
@@ -11,9 +11,9 @@ services:
       - fluence-1:/.fluence
     environment:
       RUST_BACKTRACE: "full"
-      RUST_LOG: "info,network=trace,aquamarine=info,aquamarine::actor=info,tokio_threadpool=info,tokio_reactor=info,mio=info,tokio_io=info,soketto=info,yamux=info,multistream_select=info,libp2p_secio=info,libp2p_websocket::framed=info,libp2p_ping=info,libp2p_core::upgrade::apply=info,libp2p_kad::kbucket=info,cranelift_codegen=info,wasmer_wasi=info,async_io=info,polling=info,wasmer_interface_types_fl=info
+      RUST_LOG: "info,network=trace,aquamarine=trace,aquamarine::actor=info,tokio_threadpool=info,tokio_reactor=info,mio=info,tokio_io=info,soketto=info,yamux=info,multistream_select=info,libp2p_secio=info,libp2p_websocket::framed=info,libp2p_ping=info,libp2p_core::upgrade::apply=info,libp2p_kad::kbucket=info,cranelift_codegen=info,wasmer_wasi=info,async_io=info,polling=info,wasmer_interface_types_fl=info
```
Comment (michaelvoronov) [inline deploy/fluence.yml:14]: Is it intended?

File: aquamarine/src/actor.rs [hunk 3]
```diff
@@ -131,3 +141,9 @@ impl Actor {
         }
     }
 }
+
+pub enum PollNext {
```
Comment (michaelvoronov) [inline aquamarine/src/actor.rs:145]: Maybe rename it to smth like `ActorPollResult`?

**Review by michaelvoronov — APPROVED** [review]

File: particle-node/src/futures_handle.rs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        };
+    }
+}
+
+impl Future for FuturesHandle {
+    type Output = ();
+
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let mut p_ready = false;
+        if let Some(particles) = self.particles.as_mut() {
```
[the remark this answers is not in the document]
Comment (folex) [reply particle-node/src/futures_handle.rs:62]: Maybe. I'll leave a todo

File: particle-node/src/futures_handle.rs (hunk 1 above)
Comment (folex) [reply particle-node/src/futures_handle.rs:25]: NetworkThreads?

File: particle-node/src/futures_handle.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (folex) [reply particle-node/src/futures_handle.rs]: Function will do!

File: particle-node/src/futures_handle.rs (hunk 1 above)
Comment (folex) [reply particle-node/src/futures_handle.rs]: NetworkTasks it will be

**Pull request merged**